Replace computer monitor on homepage

jvns opened the pull request:
This replaces the computer monitor on the homepage (and the links below it) with a single box.

You can see it in action here: https://jvns.github.io/git-scm.com/

I'm not totally sure about the "GitHub Repository" link, since it says elsewhere that technically the GitHub repo is a "mirror", but I think it is an official mirror and it's the most convenient way (at least for me) to view the current version of the source code.

<img width="312" height="264" alt="image" src="https://github.com/user-attachments/assets/2f068d45-22a9-4204-b36a-a2bd279b6673" />

I've removed the links below the monitor. Here's the new way to access the removed links, open to hearing that they should be made more prominent.

* Tarballs: Install -> Build from source
* Windows Build: Install -> Windows
* Mac GUIs: Tools -> GUIs

I expect this has the same issues with the screenshot not matching as #2088, haven't dealt with that yet.

Also I need to run the WCAG checker on this contrast in dark mode, not sure it's high enough.
 
<img width="194" height="38" alt="image" src="https://github.com/user-attachments/assets/9a3bda5d-e3d3-4ade-be8e-e0577fdcbbe8" />

jvns committed:
homepage: replace monitor

Co-authored-by: Marie Claire LeBlanc Flanagan <hello@marieflanagan.com>

To1ne commented on assets/sass/front-page.scss:
```diff
@@ -99,106 +99,91 @@
 
     &:hover {
-      background-image: linear-gradient(#1a7e84, #165b60);
+      transform: translateY(-1px);
```
I'm not a fan of this effect. To me it would feel more natural if the button moves down on hover. But I think I would prefer a highlighted border or something (I haven't decided yet).

jvns replied:
I can just change the background colour instead, I think that's more in line with what the other buttons on the site do

jvns replied:
It's a bit tricky to find hover background colours that work both in dark mode and light mode so I just removed the hover effect

jvns commented:
Going to try to fix the Playwright tests and merge this today.

jvns committed:
homepage: update download link tests

The name of the install link changed and many of the links were removed.
Update the tests to reflect that.

themilkman commented:
Hi there! Just a quick personal note: The design refresh really looks nice and it is good that some texts were updated. But what I don't like is that prominent "Github Repository" button below the download-button. IMHO already enough people confuse "git" with "github" (and/or think it'd be a synonym). Why not label it "Source Code" again, or "Source Code Repository" or so.

dscho commented:
@themilkman IMHO already enough people confuse open source for the thing where you tell others to implement your wishes. Why not open a Pull Request or so.

To1ne commented:
Thanks @jvns :shipit: !
